pkg/clients/edgeconnect/client_test.go

```diff
@@ -0,0 +1,246 @@
+package edgeconnect
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/oauth2"
+)
+
+const (
+	EdgeConnectOAuthClientID     = "test_client_id"
+	EdgeConnectOAuthClientSecret = "test_client_secret"
+	EdgeConnectID                = "348b4cd9-ba31-4670-9c45-9125a7d87439"
+)
+
+func TestNewClient(t *testing.T) {
+	client, err := NewClient(
+		"dummy_client_id",
+		"dummy_client_secret",
+		WithBaseURL("http://test.com"),
+		WithTokenURL("http://test.com/token"),
+	)
+	assert.NoError(t, err)
+	assert.NotNil(t, client)
+}
+
+func TestCreateEdgeConnect(t *testing.T) {
+	t.Run("create basic edge connect", func(t *testing.T) {
+		edgeConnectServer, edgeConnectClient := createTestEdgeConnectServer(t, edgeConnectCreateServerHandler(false))
+		defer edgeConnectServer.Close()
+
+		resp, err := edgeConnectClient.CreateEdgeConnect("InternalServices", []string{"*.internal.org"}, "dt0s02.AIOUP56P")
+		assert.NoError(t, err)
```

Comment:
`resp` could be nil here, right?
```suggestion
		require.NoError(t, err)
```